atlp-rwanda/stackup2-barefoot-backend
# #171380767 edit open travel request

#### What does this PR do?
Creates API endpoint used to edit travel requests
#### Description of Task to be completed?
- create API endpoint at ***PATCH/api/trips/:requestId***
- add updateTripRequest function in src/controllers/request.controller.js
- add updateTripRequest function in src/services/request.service.js
- create requesterChecker.js file in src/middlewares
- create tests to test edit travel request functionality
#### How should this be manually tested?
- After cloning the project repo, run `npm i` inside the cloned repo folder
- Make sure to rename .env.example to .env, 
- In .env file, provide all values for variables under `#Development database`  and `#General`
- Finally, run the following commands respectively:
`git fetch`
`git checkout  ft-update-open-request-171380767`
`npm i`
`npx sequelize-cli db:create`
`npx sequelize-cli db:migrate`
`npm run start`
- To edit travel request, send the following data in your request body to API at `PATCH http://<your_server_hostname>:<server_port>/api/trips/:requestId` endpoint:
```json
   "travelFrom": "kigali",
    "travelTo": "kibuye",
    "travelDate": "2020-04-30",
    "travelReason": "business",
    "travelType": "one-way",
    "accommodation": true
```
#### Any background context you want to provide?
You need to create an account and use the obtained token as an `authorization header` in your requests to the API.
#### What are the relevant pivotal tracker stories?
[#171380767](https://www.pivotaltracker.com/story/show/171380767)
#### Screenshots (if appropriate)

![Capturedi1](https://user-images.githubusercontent.com/52660542/78421064-a770a680-75f0-11ea-8664-6916f4aa9e08.PNG)

![Capturedi2](https://user-images.githubusercontent.com/52660542/78421069-af304b00-75f0-11ea-95f7-329e5134d631.PNG)

![Capturedi3](https://user-images.githubusercontent.com/52660542/78421076-b5262c00-75f0-11ea-8374-1ae4e1493331.PNG)

![Capturedi4](https://user-images.githubusercontent.com/52660542/78421081-bc4d3a00-75f0-11ea-93bc-d9a49c0cfd83.PNG)

![Capturedi5](https://user-images.githubusercontent.com/52660542/78421085-c3744800-75f0-11ea-8f12-12675670623b.PNG)
